src/dattobd.c

```diff
@@ -1785,6 +1785,157 @@ static void tp_put(struct tracing_params *tp){
 
 /****************************BIO HELPER FUNCTIONS*****************************/
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 3, 0))
+/* Patch port: block: make generic_make_request handle arbitrarily sized bios
+Ref: https://lwn.net/Articles/650246/ */
+static struct bio *snap_bio_discard_split(struct request_queue *q,
```

Comment:
this is something I should really clean up; we used the snap_prefix to indicate functions related to the 'snap_device' struct. However, those functions were written at a time when struct snap_device only handled snapshots and a different struct handled incrementals. As a result, now the 'snap_*' functions are meant to indicate that the function is related to the snap_device struct in snapshot mode. 

I would name keep the function name from the original upstream patch and use the feature detection system to avoid name conflicts.